packages/react/src/QuestionnaireBuilder/QuestionnaireBuilder.tsx

```diff
@@ -106,6 +129,18 @@ export function QuestionnaireBuilder(props: QuestionnaireBuilderProps): JSX.Elem
     };
   }, [defaultValue]);
 
+  function toggleItemOrder(itemToMove: QuestionnaireItem, direction: Direction): void {
```

Comment:
If we move this from `QuestionnaireBuilder` into `ItemBuilder`, then I think you should be able to simplify the `reorderItems` method.

Currently, `reorderItems` recursively searches for the item to move.  If you do it in `ItemBuilder`, I believe you should be able to use the array directly, similar to how we do it with `addItem` and `removeItem`